src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5565,6 +5625,44 @@ void SessionImpl::handleTorrentDeleteFailedAlert(const lt::torrent_delete_failed
     m_removingTorrents.erase(removingTorrentDataIter);
 }
 
+void SessionImpl::handleTorrentNeedCertAlert(const lt::torrent_need_cert_alert *p)
+{
+    const QString torrentID = TorrentID(p->handle.info_hash()).toString();
+
+    TorrentImpl *const torrent = m_torrents.value(p->handle.info_hash());
+    if (!torrent) [[unlikely]]
+        return;
+
+    const Path baseDir = sslPeersCertificatesDirectory();
+    if (baseDir.exists())
+    {
+
+        Path certPath = baseDir / Path(torrentID);
+        Path keyPath = certPath + u".key"_s;
+        Path dhPath = certPath + u".dh"_s;
+
+        if (!certPath.exists() || !keyPath.exists())
+        {
+            certPath = baseDir / Path(u"default"_s);
+            keyPath = baseDir / Path(u"default.key"_s);
+        }
+        if (!dhPath.exists())
```

Comment:
Why is `.dh` file handled differently? Is it valid use case when torrent uses its own cert and key files but default `.dh`?

Reply:
Yes, the DH parameters aren't actually secret. It's not problematic at all to re-use the same parameters for all torrents. So having only one parameters file, but per-torrent cert/key is a valid use-case.